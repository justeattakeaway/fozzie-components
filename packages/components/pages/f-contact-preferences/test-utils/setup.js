export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
