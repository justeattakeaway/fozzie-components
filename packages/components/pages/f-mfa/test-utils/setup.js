export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
