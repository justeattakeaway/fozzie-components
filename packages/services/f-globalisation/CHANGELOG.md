# Changelog

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

v1.0.0
------------------------------
*December 17, 2020*

### Added
- Support for `dateTimeFormats`, which requires a change in the structure of the tenancy files.


v0.2.0
------------------------------
*November 10, 2020*

### Added
- Provided a wrapper around the vue-i18n module for component tests to import


v0.1.0
------------------------------
*November 9, 2020*

### Added
- Added a Globalisation Mixin which performs all required wiring to integrate vue-i18n
