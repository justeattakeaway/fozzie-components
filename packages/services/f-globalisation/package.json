{
  "name": "@justeat/f-globalisation",
  "description": "Fozzie Globalisation â€“ A utility which wires up vue-i18n within your Smart Component",
  "version": "1.1.0",
  "main": "dist/f-globalisation.umd.min.js",
  "maxBundleSize": "15kB",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "homepage": "https://github.com/justeat/fozzie-components/tree/master/packages/services/f-globalisation",
  "contributors": [
    "Github contributors <https://github.com/justeat/fozzie-components/graphs/contributors>"
  ],
  "repository": {
    "type": "git",
    "url": "git@github.com:justeat/fozzie-components.git"
  },
  "bugs": {
    "url": "https://github.com/justeat/fozzie-components/issues"
  },
  "license": "Apache-2.0",
  "engines": {
    "node": "^12 || ^14 || ^16"
  },
  "keywords": [
    "fozzie"
  ],
  "scripts": {
    "prepublishOnly": "yarn lint && yarn test && yarn build",
    "build": "vue-cli-service build --target lib --name f-globalisation ./src/index.js",
    "lint": "eslint \"!(dist)/**/*.{js,vue}\"",
    "lint:fix": "yarn lint --fix",
    "test": "jest",
    "ci:test:services": "yarn test"
  },
  "browserslist": [
    "extends @justeat/browserslist-config-fozzie"
  ],
  "dependencies": {
    "vue-i18n": "8.28.2"
  }
}
