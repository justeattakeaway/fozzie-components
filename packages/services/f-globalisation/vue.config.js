module.exports = {
    pluginOptions: {
        lintStyleOnBuild: true
    }
};
