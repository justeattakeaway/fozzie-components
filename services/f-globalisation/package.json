{
  "name": "@justeat/f-globalisation",
  "description": "Fozzie Globalisation â€“ A utility which wires up vue-i18n within your Smart Component",
  "version": "0.2.0",
  "main": "dist/f-globalisation.umd.min.js",
  "files": [
    "dist"
  ],
  "homepage": "https://github.com/justeat/fozzie-components/tree/master/packages/f-globalisation",
  "contributors": [
    "Github contributors <https://github.com/justeat/fozzie-components/graphs/contributors>"
  ],
  "repository": {
    "type": "git",
    "url": "git@github.com:justeat/fozzie-components.git"
  },
  "bugs": {
    "url": "https://github.com/justeat/fozzie-components/issues"
  },
  "license": "Apache-2.0",
  "engines": {
    "node": ">=10.0.0"
  },
  "keywords": [
    "fozzie"
  ],
  "scripts": {
    "prepublishOnly": "yarn lint && yarn test && yarn build",
    "build": "vue-cli-service build --target lib --name f-globalisation ./src/index.js",
    "lint": "vue-cli-service lint",
    "lint:fix": "yarn lint --fix",
    "test": "vue-cli-service test:unit --json"
  },
  "browserslist": [
    "extends @justeat/browserslist-config-fozzie"
  ],
  "dependencies": {
    "vue-i18n": "8.0.0"
  },
  "devDependencies": {
    "@samhammer/vue-cli-plugin-stylelint": "2.0.1",
    "@vue/cli-plugin-babel": "4.4.6",
    "@vue/cli-plugin-eslint": "3.9.2",
    "@vue/cli-plugin-unit-jest": "4.4.6",
    "@vue/test-utils": "1.0.3",
    "vue": "2.6.10"
  }
}
