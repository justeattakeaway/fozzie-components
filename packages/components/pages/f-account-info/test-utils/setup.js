import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
