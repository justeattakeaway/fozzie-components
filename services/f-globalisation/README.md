
<div align="center">
  <h1>f-globalisation</h1>

  <img width="125" alt="Fozzie Bear" src="../../bear.png" />

  <p>A utility which wires up vue-i18n within your Smart Component</p>
</div>

---

[![npm version](https://badge.fury.io/js/%40justeat%2Ff-globalisation.svg)](https://badge.fury.io/js/%40justeat%2Ff-globalisation)
[![CircleCI](https://circleci.com/gh/justeat/fozzie-components.svg?style=svg&circle-token=4c77c1990b98c8e06e01b497bc80f376346f609d)](https://circleci.com/gh/justeat/workflows/fozzie-components)
[![Coverage Status](https://coveralls.io/repos/github/justeat/f-globalisation/badge.svg)](https://coveralls.io/github/justeat/f-globalisation)
[![Known Vulnerabilities](https://snyk.io/test/github/justeat/f-globalisation/badge.svg?targetFile=package.json)](https://snyk.io/test/github/justeat/f-globalisation?targetFile=package.json)


## Usage

1.  Install the module using NPM or Yarn:

    ```bash
    yarn add @justeat/f-globalisation
    ```

2.  Import the Mixin

    F-Globalisation contains a mixin which should be imported into your "Smart Component", for example in F-Checkout import it into the Checkout.vue component as that is the root.

    ```
    import { VueGlobalisationMixin } from '@justeat/f-globalisation';

    export default {
        mixins: [VueGlobalisationMixin]
    }
    ```

3.  Add `tenantConfigs` to your component data

    The mixin will access your localisation by accessing this data property. See F-Checkout for an example if needed. It should expose an import for the localisation files in your component.

    ```
    import tenantConfigs from '../tenants';

    export default {
        mixins: [VueGlobalisationMixin],

        data () {
            return {
                tenantConfigs
            }
        }
    }
    ```

4. Now apply vue-i18n within your components

    Once installed; you should be able to access `$t`, `<i18n>` and `this.$i18n`. The correct locale messages should be loaded automatically and `en-GB` should also be loaded as a fallback.

    vue-i18n is not needed as a dependency, because it is registered by the host application, such as CoreWeb or Storybook and thus exists in context.

    The mixin also exposes a prop; so that you can pass a locale in from a website host where required.
