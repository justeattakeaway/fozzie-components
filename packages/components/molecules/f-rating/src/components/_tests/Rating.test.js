import { createLocalVue, shallowMount } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import VRating from '../Rating.vue';
import i18n from './helpers/setup';

const localVue = createLocalVue();
localVue.use(VueI18n);

describe('Rating', () => {
    let propsData;
    let wrapper;
    const $tc = jest.fn();

    beforeEach(() => {
        propsData = {
            starRating: 2,
            maxStarRating: 5
        };
        wrapper = shallowMount(VRating, {
            propsData,
            localVue,
            i18n,
            mocks: {
                $tc
            }
        });
    });

    it('should be defined', () => {
        expect(wrapper.exists()).toBe(true);
    });

    describe('methods', () => {
        describe('`isRatingStarFilled`', () => {
            it('should exist', () => {
                expect(wrapper.vm.isRatingStarFilled).toBeDefined();
            });

            it('should contain a description `c-rating-description`', () => {
                // Act
                const result = wrapper.find('[data-test-id="c-rating-description"]');

                // Assert
                expect(result).toMatchSnapshot();
            });

            describe('when invoked', () => {
                it('should return truthy when the argument `star` is less than or equal to `starRating`', () => {
                    // Act
                    const star = 1;
                    const result = wrapper.vm.isRatingStarFilled(star);

                    // Assert
                    expect(result).toBe(true);
                });

                it.each([1, 2])('should return truthy when the argument star is %s', value => {
                    // Arrange
                    propsData = {
                        starRating: value,
                        maxStarRating: 5
                    };

                    // Act
                    wrapper = shallowMount(VRating, {
                        propsData,
                        localVue,
                        i18n,
                        mocks: {
                            $tc
                        }
                    });
                    const result = wrapper.vm.isRatingStarFilled(value);

                    // Assert
                    expect(result).toBe(true);
                });

                it('should return truthy when the argument `star` is equal to `starRating`', () => {
                    // Act
                    const star = 2;
                    const result = wrapper.vm.isRatingStarFilled(star);

                    // Assert
                    expect(result).toBe(true);
                });

                it('should return falsey when argument `star` is greater than `starRating`', () => {
                    // Act
                    const star = 3;
                    const result = wrapper.vm.isRatingStarFilled(star);

                    // Assert
                    expect(result).toBe(false);
                });
            });
        });
    });

    describe('computed', () => {
        describe('`getRatingDescription`', () => {
            it('should exist', () => {
                // Arrange
                propsData.starRating = 2;
                wrapper = shallowMount(VRating, {
                    propsData,
                    localVue,
                    i18n
                });

                // Act & Assert
                expect(wrapper.vm.getRatingDescription).toBeDefined();
            });

            describe('when invoked', () => {
                it('should return a singular description if the rating is less than 2', () => {
                    // Arrange
                    propsData = {
                        starRating: 1
                    };
                    wrapper = shallowMount(VRating, {
                        propsData,
                        localVue,
                        i18n
                    });

                    // Act & Assert
                    expect(wrapper.vm.getRatingDescription).toMatchSnapshot();
                });

                it('should return a plural description if the rating is greater than 1', () => {
                    // Arrange
                    propsData = {
                        starRating: 2
                    };
                    wrapper = shallowMount(VRating, {
                        propsData,
                        localVue,
                        i18n
                    });

                    // Act & Assert
                    expect(wrapper.vm.getRatingDescription).toMatchSnapshot();
                });
            });
        });
    });
});
