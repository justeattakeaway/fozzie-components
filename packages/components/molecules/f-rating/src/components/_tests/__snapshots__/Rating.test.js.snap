// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating computed \`getRatingDescription\` when invoked should return a plural description if the rating is greater than 1 1`] = `"2 stars out of 5"`;

exports[`Rating computed \`getRatingDescription\` when invoked should return a singular description if the rating is less than 2 1`] = `"1 star out of 5"`;
